### PR TITLE
Feat(eos_cli_config_gen): Enhance support for PTP monitoring

### PR DESCRIPTION
**Pull request by nielsjlarsen**
## Change Summary

Updated PTP config to reflect all necessary knobs needed to support PR #1916.

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Added support for PTP monitor thresholds for missing messages:

```yml
ptp:
  monitor:
    enabled: < true | false | default -> true >
    threshold:
      offset_from_master: < 0-1000000000 >
      mean_path_delay: < 0-1000000000 >
      drop:
        offset_from_master: < 0-1000000000 >
        mean_path_delay: < 0-1000000000 >
    missing_message:
      intervals:
        announce: < 2-255 >
        follow_up: < 2-255 >
        sync: < 2-255 >
      sequence_ids:
        enabled: < true | false | default -> false >
        announce: < 2-255 >
        delay_resp: < 2-255 >
        follow_up: < 2-255 >
        sync: < 2-255 >
```

## How to test

molecule converge --scenario-name eos_cli_config_gen --    

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -2654,24 +2654,29 @@ patch_panel:
...
+  priority2: < integer >
+  ttl: < 1-254 >
+  domain: < 0-255 >
   message_type:
     general:
       dscp: < dscp-value >
     event:
-      dscp: < dscp-Value >
+      dscp: < dscp-value >
   monitor:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:2673]: please add
```
ptp:
  monitor:
    enabled: < true | false >
    sequence_id: < true | false >
```
Sequence_id is required (eos default off) to support the thresholds you are adding.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 2]
```diff
@@ -2654,24 +2654,29 @@ patch_panel:
...
     threshold:
-      offset_from_master: < offset >
-      mean_path_delay: < delay >
+      offset_from_master: < integer >
+      mean_path_delay: < integer >
+      missing_message:
+        sync: < integer >
+        follow_up: < integer >
+        delay_resp: < integer >
+        announce: < integer >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: In the CLI each of these options must have 2-255 intervals and/or 2-255 sequence-ids
Suggestion
```
        missing_message:
          sync:
            intervals: < 2-255 >
            sequence_ids: < 2-255 >
        ...
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 3]
```diff
@@ -2654,24 +2654,29 @@ patch_panel:
...
       dscp: < dscp-value >
     event:
-      dscp: < dscp-Value >
+      dscp: < dscp-value >
   monitor:
     threshold:
-      offset_from_master: < offset >
-      mean_path_delay: < delay >
+      offset_from_master: < integer >
+      mean_path_delay: < integer >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: This is not an integer. Takes `0-1000000000 nanoseconds`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (nielsjlarsen) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:2673]: Done

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 2 above)
Comment (nielsjlarsen) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Updated

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 3 above)
Comment (nielsjlarsen) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Updated

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 4]
```diff
@@ -2655,24 +2655,41 @@ patch_panel:
...
-  clock_identity: < clock-id >
+  clock_identity: < "clock-id in xx:xx:xx:xx:xx:xx format" >
   source:
-    ip: < source-ip>
-  priority1: < priority1 >
-  priority2: < priority2 >
-  ttl: < ttl >
-  domain: < integer >
+    ip: < source-ip >
+  priority1: < integer >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: 0-255 - same for pri2

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 5]
```diff
@@ -2655,24 +2655,41 @@ patch_panel:
...
+  ttl: < 1-254 >
+  domain: < 0-255 >
   message_type:
     general:
       dscp: < dscp-value >
     event:
-      dscp: < dscp-Value >
+      dscp: < dscp-value >
   monitor:
+    enabled: < false | true -> default >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: To be consistent with the rest of AVD. Same below
```suggestion
    enabled: < false | true | default -> true >
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 6]
```diff
@@ -2655,24 +2655,41 @@ patch_panel:
...
+        offset_from_master: < 0-1000000000 >
+        mean_path_delay: < 0-1000000000 >
+    missing_message:
+      intervals:
+        enabled: < true | false -> default >
+        announce: < 2-255 >
+        follow_up: < 2-255 >
+        sync: < 2-255 >
+      sequence_ids:
+        enabled: < true | false -> default >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: This does not follow the cli model. The enable for sequence ids is directly under monitor. In eos_cli_config_gen we should try to match the cli model, and not create "clever" abstraction :)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 7]
```diff
@@ -2655,24 +2655,41 @@ patch_panel:
...
+      dscp: < dscp-value >
   monitor:
+    enabled: < false | true -> default >
     threshold:
-      offset_from_master: < offset >
-      mean_path_delay: < delay >
+      offset_from_master: < 0-1000000000 >
+      mean_path_delay: < 0-1000000000 >
+      drop:
+        enabled: < true | false -> default >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: this should not be needed and has no 1:1 relation with the CLI. Just detect on the threshold number and submit the CLI.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 [hunk 8]
```diff
@@ -31,10 +31,52 @@ ptp mode {{ ptp.mode }}
 {%     if ptp.forward_unicast is arista.avd.defined(true) %}
 ptp forward-unicast
 {%     endif %}
-{%     if ptp.monitor.threshold.offset_from_master is arista.avd.defined %}
-ptp monitor threshold offset-from-master {{ ptp.monitor.threshold.offset_from_master }}
+{%     if ptp.monitor.enabled is arista.avd.defined(false) %}
+no ptp monitor
 {%     endif %}
-{%     if ptp.monitor.threshold.mean_path_delay is arista.avd.defined %}
+{%     if ptp.monitor.enabled is not arista.avd.defined(false) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2]: Replace this endif + new if with 
`{%     elif ptp.monitor is arista.avd.defined %}`
So we still default to `true` but we don't have to traverse all this code for every device unless they have ptp.monitor defined.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 [hunk 9]
```diff
@@ -31,10 +31,52 @@ ptp mode {{ ptp.mode }}
...
+ptp monitor threshold missing-message follow-up {{ ptp.monitor.missing_message.intervals.follow_up }} intervals
+{%             endif %}
+{%             if ptp.monitor.missing_message.intervals.sync is arista.avd.defined %}
+ptp monitor threshold missing-message sync {{ ptp.monitor.missing_message.intervals.sync }} intervals
+{%             endif %}
+{%         endif %}
+{%         if ptp.monitor.missing_message.sequence_ids.enabled is arista.avd.defined(false) %}
+no ptp monitor sequence-id
+{%         endif %}
+{%         if ptp.monitor.missing_message.sequence_ids.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2]:
```suggestion
{%         elif ptp.monitor.missing_message.sequence_ids.enabled is arista.avd.defined(true) %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 [hunk 10]
```diff
@@ -31,10 +31,52 @@ ptp mode {{ ptp.mode }}
...
 {%     endif %}
-{%     if ptp.monitor.threshold.mean_path_delay is arista.avd.defined %}
+{%     if ptp.monitor.enabled is not arista.avd.defined(false) %}
+{%         if ptp.monitor.threshold.offset_from_master is arista.avd.defined %}
+ptp monitor threshold offset-from-master {{ ptp.monitor.threshold.offset_from_master }}
+{%         endif %}
+{%         if ptp.monitor.threshold.mean_path_delay is arista.avd.defined %}
 ptp monitor threshold mean-path-delay {{ ptp.monitor.threshold.mean_path_delay }}
+{%         endif %}
+{%         if ptp.monitor.threshold.drop.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2]:
```suggestion
{%         if ptp.monitor.threshold.drop is arista.avd.defined %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 [hunk 11]
```diff
@@ -31,10 +31,52 @@ ptp mode {{ ptp.mode }}
...
+{%             if ptp.monitor.threshold.drop.offset_from_master is arista.avd.defined %}
+ptp monitor threshold offset-from-master {{ ptp.monitor.threshold.drop.offset_from_master }} nanoseconds drop
+{%             endif %}
+{%             if ptp.monitor.threshold.drop.mean_path_delay is arista.avd.defined %}
+ptp monitor threshold mean-path-delay {{ ptp.monitor.threshold.drop.mean_path_delay }} nanoseconds drop
+{%             endif %}
+{%         endif %}
+{%         if ptp.monitor.missing_message.intervals.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2]:
```suggestion
{%         if ptp.monitor.missing_message.intervals is arista.avd.defined %}
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 6 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: TBF this makes a lot of sense - more than the CLI :) I think we can keep this one.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 12]
```diff
@@ -2655,24 +2655,41 @@ patch_panel:
...
+      dscp: < dscp-value >
   monitor:
+    enabled: < false | true | default -> true >
     threshold:
-      offset_from_master: < offset >
-      mean_path_delay: < delay >
+      offset_from_master: < 0-1000000000 >
+      mean_path_delay: < 0-1000000000 >
+      drop:
+        # enabled: < true | false -> default >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 13]
```diff
@@ -2655,24 +2655,41 @@ patch_panel:
...
+        offset_from_master: < 0-1000000000 >
+        mean_path_delay: < 0-1000000000 >
+    missing_message:
+      intervals:
+        enabled: < true | false -> default >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]:
```suggestion
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml [hunk 14]
```diff
@@ -1,24 +1,41 @@
...
+    threshold:
+      offset_from_master: 11
+      mean_path_delay: 12
+      drop:
+        enabled: true
+        offset_from_master: 13
+        mean_path_delay: 14
+    missing_message:
+      intervals:
+        enabled: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml]:
```suggestion
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml [hunk 15]
```diff
@@ -1,24 +1,41 @@
...
+      dscp: 46
   mode: boundary
   forward_unicast: true
+  monitor:
+    # enabled: true (default)
+    threshold:
+      offset_from_master: 11
+      mean_path_delay: 12
+      drop:
+        enabled: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 [hunk 16]
```diff
@@ -31,10 +31,48 @@ ptp mode {{ ptp.mode }}
...
+{%             if ptp.monitor.missing_message.intervals.follow_up is arista.avd.defined %}
+ptp monitor threshold missing-message follow-up {{ ptp.monitor.missing_message.intervals.follow_up }} intervals
+{%             endif %}
+{%             if ptp.monitor.missing_message.intervals.sync is arista.avd.defined %}
+ptp monitor threshold missing-message sync {{ ptp.monitor.missing_message.intervals.sync }} intervals
+{%             endif %}
+{%         endif %}
+{%         if ptp.monitor.missing_message.sequence_ids.enabled is arista.avd.defined(false) %}
+no ptp monitor sequence-id
+{%         elif ptp.monitor.missing_message.sequence_ids is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2]:
```suggestion
{%         elif ptp.monitor.missing_message.sequence_ids.enabled is arista.avd.defined(true) %}
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2 [hunk 17]
```diff
@@ -4,40 +4,17 @@
...
-{%             endif %}
-{%         endif %}
-{%         if ptp.message_type.event is defined and ptp.message_type.event is not none %}
-{%             if ptp.message_type.event.dscp is defined and ptp.message_type.event.dscp is not none %}
-| Msg Event | DSCP {{ ptp.message_type.event.dscp }} |
-{%             endif %}
-{%         endif %}
-{%     endif %}
+| Clock ID    | Source IP | Priority 1 | Priority 2 | TTL | Domain | Mode        | Forward Unicast |
+| ----------- | --------- | ---------- | ---------- | --- | ------ | ----------- | --------------- |
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2]:
```suggestion
| Clock ID | Source IP | Priority 1 | Priority 2 | TTL | Domain | Mode | Forward Unicast |
| -------- | --------- | ---------- | ---------- | --- | ------ | ---- | --------------- |
```

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**